feat: Align ERC‑8021 Schema 1 serialization, deserialization with EIP‑8021 (variable‑length chainId)

wbj-cb opened the pull request:
<!-- What is this PR solving? Write a clear description or reference the issues it solves (e.g. `fixes #123`). What other alternatives have you explored? Are there any parts you think require more attention from reviewers? -->

# What
* Updates `Attribution.fromData` to deserialize the variable length registry section from schema 1
* Updates `Attribution.toDataSuffix` to serialize the variable length registry section when schema 1
* Updates tests and adds additional tests cases to verify round tripping results in the same payload

# Why:
EIP‑8021 changed to have variable‑length chainId for Schema 1:

> Schema ID 1 builds on top of 0, but also defines a custom Code Registry chain and address
>
> 1. codesLength (1 byte): Length of the codes array in bytes, including delimiters
> 2. codes (variable length): Array of attribution codes using ASCII encoding with comma (0x2C) as delimiter
> 3. **codeRegistryChainIdLength (1 byte): Length of the chain id hosting the custom Code Registry**
> 4. **codeRegistryChainId (variable bytes): Chain id hosting the custom Code Registry**
> 5. **codeRegistryAddress (20 bytes): Address of a smart contract implementing ICodeRegistry**
> Custom Code Registries can be referenced across chains for ease of connecting identity for multi-chain apps.

From https://eip.tools/eip/8021


<!----------------------------------------------------------------------
Before creating the pull request, please make sure you do the following:

- Read the Contributing Guidelines at https://github.com/wevm/ox/blob/main/.github/CONTRIBUTING.md
- Check that there isn't already a PR that solves the problem the same way. If you find a duplicate, please help us review it.
- Update the corresponding documentation if needed.
- Include relevant tests that fail without this PR, but pass with it.

Thank you for contributing to Ox!
----------------------------------------------------------------------->

tmm commented on src/erc8021/Attribution.ts:
```diff
@@ -31,12 +31,20 @@ export type AttributionSchemaId0 = {
-  /** Address of the custom code registry contract. */
-  codeRegistryAddress: Address.Address
+  /* The custom code registry contract. */
+  codeRegistry: AttributionScehamId1Registry
```
```suggestion
  codeRegistry: AttributionSchemaId1Registry
```

tmm commented on src/erc8021/Attribution.ts:
```diff
@@ -31,12 +31,20 @@ export type AttributionSchemaId0 = {
+export type AttributionScehamId1Registry = {
+  /** Address of the custom code registry contract. */
+  address: Address.Address
+
```
```suggestion
```

wbj-cb commented:
@tmm @jxom Addressed your comments

* Fixed mispelled `AttributionScehamId1Registry` -> `AttributionSchemaId1Registry`
* Updated changeset from `patch` to `major` because of breaking API changes
* Updated docstring examples to reflect changes
